{"url":"https://api.github.com/repos/elastic/elasticsearch/issues/16793","repository_url":"https://api.github.com/repos/elastic/elasticsearch","labels_url":"https://api.github.com/repos/elastic/elasticsearch/issues/16793/labels{/name}","comments_url":"https://api.github.com/repos/elastic/elasticsearch/issues/16793/comments","events_url":"https://api.github.com/repos/elastic/elasticsearch/issues/16793/events","html_url":"https://github.com/elastic/elasticsearch/issues/16793","id":136026528,"node_id":"MDU6SXNzdWUxMzYwMjY1Mjg=","number":16793,"title":"ES 2.2.0 wrongfully marks shard as unassigned / corrupted","user":{"login":"jrots","id":195346,"node_id":"MDQ6VXNlcjE5NTM0Ng==","avatar_url":"https://avatars1.githubusercontent.com/u/195346?v=4","gravatar_id":"","url":"https://api.github.com/users/jrots","html_url":"https://github.com/jrots","followers_url":"https://api.github.com/users/jrots/followers","following_url":"https://api.github.com/users/jrots/following{/other_user}","gists_url":"https://api.github.com/users/jrots/gists{/gist_id}","starred_url":"https://api.github.com/users/jrots/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/jrots/subscriptions","organizations_url":"https://api.github.com/users/jrots/orgs","repos_url":"https://api.github.com/users/jrots/repos","events_url":"https://api.github.com/users/jrots/events{/privacy}","received_events_url":"https://api.github.com/users/jrots/received_events","type":"User","site_admin":false},"labels":[{"id":73544,"node_id":"MDU6TGFiZWw3MzU0NA==","url":"https://api.github.com/repos/elastic/elasticsearch/labels/%3Enon-issue","name":">non-issue","color":"cfcfcf","default":false,"description":null}],"state":"closed","locked":false,"assignee":{"login":"s1monw","id":973334,"node_id":"MDQ6VXNlcjk3MzMzNA==","avatar_url":"https://avatars0.githubusercontent.com/u/973334?v=4","gravatar_id":"","url":"https://api.github.com/users/s1monw","html_url":"https://github.com/s1monw","followers_url":"https://api.github.com/users/s1monw/followers","following_url":"https://api.github.com/users/s1monw/following{/other_user}","gists_url":"https://api.github.com/users/s1monw/gists{/gist_id}","starred_url":"https://api.github.com/users/s1monw/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/s1monw/subscriptions","organizations_url":"https://api.github.com/users/s1monw/orgs","repos_url":"https://api.github.com/users/s1monw/repos","events_url":"https://api.github.com/users/s1monw/events{/privacy}","received_events_url":"https://api.github.com/users/s1monw/received_events","type":"User","site_admin":false},"assignees":[{"login":"s1monw","id":973334,"node_id":"MDQ6VXNlcjk3MzMzNA==","avatar_url":"https://avatars0.githubusercontent.com/u/973334?v=4","gravatar_id":"","url":"https://api.github.com/users/s1monw","html_url":"https://github.com/s1monw","followers_url":"https://api.github.com/users/s1monw/followers","following_url":"https://api.github.com/users/s1monw/following{/other_user}","gists_url":"https://api.github.com/users/s1monw/gists{/gist_id}","starred_url":"https://api.github.com/users/s1monw/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/s1monw/subscriptions","organizations_url":"https://api.github.com/users/s1monw/orgs","repos_url":"https://api.github.com/users/s1monw/repos","events_url":"https://api.github.com/users/s1monw/events{/privacy}","received_events_url":"https://api.github.com/users/s1monw/received_events","type":"User","site_admin":false}],"milestone":null,"comments":13,"created_at":"2016-02-24T11:03:39Z","updated_at":"2016-03-02T20:44:00Z","closed_at":"2016-03-02T19:38:13Z","author_association":"NONE","active_lock_reason":null,"body":"Hi I had a red state on one of the shards in my index : \n\nI tried a force relocation to reassign it, this was the output:\nhttps://gist.github.com/jrots/da71bc8b0ad86a27636a\nforce allocation from previous reason ALLOCATION_FAILED, failed recovery,\n failure IndexShardRecoveryException[failed to fetch index version after copying it over]; nested: \n CorruptIndexException[failed engine (reason: [search execution corruption failure]) (resource=preexisting_corruption)]; nested: NotSerializableExceptionWrapper[failed engine (reason: [search execution corruption failure])]; \n nested: CorruptIndexException[Corrupted: docID=3412736, docBase=53, chunkDocs=50, numDocs=4460365\nThere was indeed a \"corrupted_..\" file present, \n\nI did a check with lucene CheckIndex, this was the output : \nhttps://gist.github.com/jrots/e816c24b17ef4ae6fe29\n\nall segments showed up as valid...  \n\nI manually removed the corrupt_ file and did a restart of that node & all was assigned again and ES had a green state again.. but there might be a bug in ES that generates a \"corrupted_\" file too soon because there weren't any issues when doing a check afterwards.. \n","closed_by":{"login":"s1monw","id":973334,"node_id":"MDQ6VXNlcjk3MzMzNA==","avatar_url":"https://avatars0.githubusercontent.com/u/973334?v=4","gravatar_id":"","url":"https://api.github.com/users/s1monw","html_url":"https://github.com/s1monw","followers_url":"https://api.github.com/users/s1monw/followers","following_url":"https://api.github.com/users/s1monw/following{/other_user}","gists_url":"https://api.github.com/users/s1monw/gists{/gist_id}","starred_url":"https://api.github.com/users/s1monw/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/s1monw/subscriptions","organizations_url":"https://api.github.com/users/s1monw/orgs","repos_url":"https://api.github.com/users/s1monw/repos","events_url":"https://api.github.com/users/s1monw/events{/privacy}","received_events_url":"https://api.github.com/users/s1monw/received_events","type":"User","site_admin":false},"performed_via_github_app":null}